scopes/envs/envs/envs.docs.md

```diff
@@ -1,33 +1,797 @@
 
-Also, it is recommended to implement the following:
+> As with any other 'merging' process, the properties defined in the above returned object will be added to configurations set by Bit.  
+> Conflicting properties will be overridden by the properties that are set here.  
+> Configurations that are set here may also be overridden, either by the [Dependency Resolver aspect](https://bit.dev/teambit/dependencies/dependency-resolver) or by workspace configurations set using the [variants API](https://bit.dev/teambit/workspace/variants).
+
+#### getBuildPipe
+
+```ts
+getBuildPipe(...args : any[]): BuildTask[]
 ```
-getCompiler(): Compiler; // if you need compiler
-getTester(): Tester; // if you need tester
-getDependencies(); // if you need to change/add/remove package dependencies
+
+Returns an array of build tasks to be used by the Builder service. Tasks will be added after and before Bit's pre-configured build tasks. Learn more about it [here](/docs/build-pipeline/overview).
+
+For example:
+
+```ts
+export class ReactEnv implements Environment {
+  constructor(
+    // ...
+
+    // The Compiler aspect
+    private compiler: CompilerMain,
+
+    // The Tester aspect
+    private tester: TesterMain
+  ) {}
+
+  getBuildPipe(): BuildTask[] {
+    return [
+      this.compiler.createTask('StencilCompiler', this.getCompiler()),
+      this.tester.task,
+    ];
+  }
+}
 ```
 
-# Configure workspace to use the env
+## Extending multiple runtime environments 
+
+An environment may operate in multiple runtime environments: 'Main', which runs on the server and 'UI' and 'Preview', which run on the browser.
+Each runtime environment runs all files that are named with its corresponding file pattern.
+
+An environment extension that runs on multiple runtimes is called "Aspect" an will have the following file structure:
 
-add this to your workspace.jsonc file.
 ```
-"teambit.envs/envs": {
-  "env": "your-new-env",
-  "config": {}
+|-- env-extension
+    |-- env-extension.main.ts
+    |-- env-extension.ui.tsx
+    |-- env-extension.preview.tsx
+    |-- env.extension.aspect.ts
+```
+
+### Registering an environment as an aspect
+
+Create a `*.aspect.ts` file:
+
+For example:
+```shell
+$ touch path/to/extension/env-extension.aspect.ts
+```
+
+Place the following lines to register your environment as a multiple runtime extension (a.k.a, an Aspect):
+
+```ts
+// env-extension.ts
+
+import { Aspect } from '@teambit/harmony';
+
+export const ReactWithProvidersAspect = Aspect.create({
+  // The ID should be your component's ID
+  // Make sure to track your extension component before registering it as an Aspect
+  id: 'my-scope.react-with-providers',
+});
+
+```
+
+### Registering a runtime extension
+An aspect is a collection of multiple extensions, each extending a specific runtime.
+
+Register each runtime extension to its corresponding runtime, using the `addRuntime` method.
+
+For example:
+
+```typescript
+// react-with-providers.preview.ts
```

Comment:
inconsistent with the example above - 
above you used env-extesnion.ts in the comment and here the react-with-providers
make all consistent

Reply:
I aimed for two independent examples but I can see how that can be confusing. (Done.)